### ISSUE TITLE
IDE0055 false positive for C# 14 extension methods

### ISSUE THREAD

**Issue by znakeeye** [dotnet/roslyn#78422]
Targeting `.NET 10` with `preview` version of `C#`. The following code (produced by code-fix) yields IDE0055 warning. It seems to dislike the `extension` keyword...

```c#
internal static class StringExtensions
{
    extension(string instance)
    {
        public bool L33t() => instance.Length == 1338;
    }
}
```

Sample project: [Ide0055.zip](https://github.com/user-attachments/files/20019295/Ide0055.zip)

Relates to extensions (test plan https://github.com/dotnet/roslyn/issues/76130)

**Comment by CyrusNajmabadi** [comment]
I don't repro this on latest preview bits.  This was likely reported before any IDE work had been done for extensions.
